package.json

```diff
@@ -38,19 +45,54 @@
 		"rules": {
 			"no-implicit-globals": "off",
 			"@typescript-eslint/prefer-nullish-coalescing": "off",
-			"import/no-unassigned-import": "off"
+			"import/extensions": "off",
+			"import/no-unassigned-import": "off",
+			"n/file-extension-in-import": "off"
 		}
 	},
+	"jest": {
+		"injectGlobals": false,
+		"preset": "jest-puppeteer",
+		"testMatch": [
+			"**/test/*.js"
+		]
+	},
 	"dependencies": {
 		"content-scripts-register-polyfill": "^3.2.2",
 		"webext-additional-permissions": "^2.3.0",
 		"webext-content-scripts": "^1.0.2"
 	},
 	"devDependencies": {
+		"@parcel/config-webextension": "^2.6.0",
 		"@sindresorhus/tsconfig": "^3.0.1",
 		"@types/chrome": "^0.0.190",
 		"@types/firefox-webext-browser": "^94.0.1",
+		"jest": "^28.1.1",
+		"jest-puppeteer": "^6.1.0",
+		"parcel": "^2.6.0",
+		"puppeteer": "^14.4.0",
 		"typescript": "^4.7.3",
 		"xo": "^0.50.0"
+	},
+	"targets": {
+		"main": false,
+		"module": false,
+		"default": {
+			"source": "./test/demo-extension/manifest.json",
+			"distDir": "./test/dist",
+			"sourceMap": {
+				"inline": true
+			},
+			"optimize": false
+		}
+	},
+	"webExt": {
+		"sourceDir": "test/dist",
+		"run": {
+			"startUrl": [
+				"https://iframe-test-page.vercel.app/",
+				"https://iframe-test-page.vercel.app/cross-origin-parent.html"
```

Comment:
Potentially out of date. Same in https://github.com/fregante/content-scripts-register-polyfill/blob/main/test/test.js